### PR TITLE
Experimental TextmeshPro and UnityUI input field helpers that work with HoloLens keyboard

### PR DESCRIPTION
**Pull request by julenka**
**This PR is a continuation of the PR from @Alexees #6036, thank you to alexees for the original fix.**

Additional changes on top of @Alexees original PR: 
- Moves files related to the MixedRealityKeyboard to SDK/Experimental/MixedRealityKeyboard/
- Move MixedRealityKeyboardPreview classes into Experimental/MixedRealityKeyboardPreview folder
- Add text input field to the hand interaction examples scene

Original PR from @Alexees:

## Overview
The class MixedRealityKeyboard is a MonoBehaviour that shows how to Open, Close and use the system native keyboards on UWP platforms.

The programmer still has the responsibility to link it with whatever she/he needs the input for. This means she/he needs to write additional MonoBehaviour scripts to link the with for example InputFields, which activate on click.

I wanted to make this easier in that InputFields and TMP_InputFields are handled directly by just one additional script attached to them.

Fixes: #3970 by providing a workaround -- to get the keyboard to show up, attach the input field helper component to the gameobject containing the input field.

## Changes
This set of scripts provides the following:
- A script, once attached to Unity InputFields, automatically manages the native keyboard and updates the InputFields text
- Same for the TMP_InputField
- Base classes to derive from if you want to use the keyboard for something other than InputFields

I first went for overriding the InputFields, but first, they are far too complicated to just do that and second, The existing implementation of MixedRealityKeyboard.cs already completely ignores InputField interactions and second there might be more uses for the keyboard than just InputFields.

### CODE REVIEW

File: Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/KeyboardInputFieldBase.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using UnityEngine;
+using UnityEngine.EventSystems;
+using Microsoft.MixedReality.Toolkit.Input;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// Base class explicitly launching Windows Mixed Reality's system keyboard for InputField and TMP_InputField
+    /// To be attached to the same GameObject with either of the components
```
Comment (julenka) [inline Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/KeyboardInputFieldBase.cs]:
```suggestion
    /// To be attached to the same GameObject with either of the components.
```

**Comment by julenka** [comment]
@cre8ivepark FYI, once this PR is in we will have a way for Unity UI text input fields to work well with the Hololens 2 system keyboard. I noticed that the Unity UI sample in HandInteractionExamples does not include text input fields. We can add text input fields to the examples after this PR gets in.

**Comment by julenka** [comment]
Found a bug:  after typing text and closing keyboard, then re-opening the keyboard, text will be set to empty string and you see "enter text..." prompt again.

**Comment by CDiaz-MS** [comment]
Found another bug:
In the UI.InputField box if you enter text, close the keyboard, enter text again, close the keyboard, then try to enter text, the input box does not reflect the keys pressed on the keyboard.

Side note:  The keyboard does not work while remoting only when you build and deploy.

File: Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs [hunk 2]
```diff
@@ -1,38 +1,25 @@
+//#define WINDOWS_UWP
```
Comment (davidkline-ms) [inline Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs]: Why is this line here? It is commented, so is safe, but the code should get platform defines from the project.

Comment (julenka) [reply Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/MixedRealityKeyboardBase.cs]: Good catch!

**Comment by julenka** [comment]
> Found a bug: after typing text and closing keyboard, then re-opening the keyboard, text will be set to empty string and you see "enter text..." prompt again.

Fixed

**Comment by julenka** [comment]
> Found another bug:
> In the UI.InputField box if you enter text, close the keyboard, enter text again, close the keyboard, then try to enter text, the input box does not reflect the keys pressed on the keyboard.

After my latest fix I can't reproduce the issue, seems fixed, or not 100% repro. Since this is experimental, bugs that occur infrequently (< 25% of cases) are acceptable. I noticed in one of four times I brought up keyboard, that I typed "One" and the letters ended up "noo", but was not able to reproduce after several more attempts.

File: Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/README_MixedRealityKeyboard.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Mixed Reality and HoloLens Keyboard helpers
+
+MRTK provides several experimental helper components to assist with launching and reading text from the [System Keyboard](../../../../../Documentation/README_SystemKeyboard.md). 
+
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/README_MixedRealityKeyboard.md]: It might be helpful to note that the keyboard does not work in editor or while remoting.

Comment (julenka) [reply Assets/MRTK/SDK/Experimental/MixedRealityKeyboard/README_MixedRealityKeyboard.md]: Done.

**Comment by julenka** [comment]
For the future: @CDiaz-MS noticed that Unity UI will sometimes select all of the text when the text input field is selected. Since this is the default Unity behavior, we decided not to change it. If we want to force Unity UI to always move the caret to the end, solutions are presented here: https://stackoverflow.com/questions/41484725/how-to-disable-selectall-text-of-ui-inputfield-when-it-gets-focus

File: Documentation/ReleaseNotes.md [hunk 4]
```diff
@@ -52,6 +52,13 @@ This version of MRTK modifies the layout of the MRTK folder structure. This chan
 
 This version of MRTK adds three new methods to the `WindowsApiChecker` class: `IsMethodAvailable`, `IsPropertyAvailable` and `IsTypeAvailable`. These methods allow for checking for feature support on Windows 10 and are prefered over using the `UniversalApiContractV#_IsAvailable` properties.
 
+**Helpers to get text input fields working with MixedRealityKeyboard for UnityUI, TextMeshPro (Experimental)**
+
```
Comment (julenka) [inline Documentation/ReleaseNotes.md]:
```suggestion
![image](https://user-images.githubusercontent.com/168492/77582981-86e07800-6e9d-11ea-86e5-bf2c0840296c.png)

```
